pilot/pkg/xds/ads.go

```diff
@@ -482,12 +482,11 @@ func (s *DiscoveryServer) initConnection(node *core.Node, con *Connection) error
 	}
 
 	// First request so initialize connection id and start tracking it.
-	con.mu.Lock()
```

Comment:
We donot need this lock, it is initiated here will not be used before adding to connection pool